app/src/main/kotlin/net/syncthing/lite/library/DefaultLibraryManager.kt

```diff
@@ -0,0 +1,104 @@
+package net.syncthing.lite.library
+
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import net.syncthing.lite.BuildConfig
+import net.syncthing.lite.R
+import org.jetbrains.anko.defaultSharedPreferences
+
+object DefaultLibraryManager {
+    private const val LOG_TAG = "DefaultLibraryManager"
+    private const val COUNTDOWN_STEP_SIZE = 1000L
+
+    private var instance: LibraryManager? = null
+    private val lock = Object()
+    private val handler = Handler(Looper.getMainLooper())
+
+    fun with(context: Context) = withApplicationContext(context.applicationContext)
+
+    private fun withApplicationContext(context: Context): LibraryManager {
+        if (instance == null) {
+            synchronized(lock) {
+                if (instance == null) {
+                    var isRunning = false
+                    var isUsed = false
+                    var unusedTime = 0L
+                    lateinit var countDownStep: Runnable
+
+                    fun scheduleCountdownStep() {
+                        handler.removeCallbacks(countDownStep)
+                        handler.postDelayed(countDownStep, COUNTDOWN_STEP_SIZE)
+                    }
+
+                    fun cancelCountdown() {
+                        unusedTime = 0
+                        handler.removeCallbacks(countDownStep)
+                    }
+
+                    countDownStep = Runnable {
+                        unusedTime += COUNTDOWN_STEP_SIZE
+
+                        val shutdownDelay = context.defaultSharedPreferences.getString(
+                                "shutdown_delay",
+                                context.getString(R.string.default_shutdown_delay)
+                        ).toLong()
+
+                        if (unusedTime >= shutdownDelay) {
+                            instance!!.shutdownIfThereAreZeroUsers {
+                                // ignore the result, we are informed using the isRunningListener too
+                            }
+                        } else {
+                            // update notification
+                            LibraryConnectionService.notifyRunningAndUnused((shutdownDelay - unusedTime) / 1000, context)
```

Comment:
This uses an Handler and not an AlarmManager. This means that the device will not wake up from standby to execute this. Moreover, there is nowhere a WakeLock (not even the permission to keep the device awake, see <https://github.com/syncthing/syncthing-lite/blob/master/app/src/main/AndroidManifest.xml>) so that the application will be pauses as soon as nothing else keeps the device active (normally when the screen is off and no charger is connected).

Getting the index of all connected devices all 15 seconds <https://github.com/syncthing/syncthing-java/blob/master/client/src/main/kotlin/net/syncthing/java/client/SyncthingClient.kt#L54> is not good for the battery, but the same rules apply here.